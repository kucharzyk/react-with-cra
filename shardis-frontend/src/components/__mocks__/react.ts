import * as React from 'react';

// Temporary hack to suppress error
// https://github.com/facebookincubator/create-react-app/issues/3199
window.requestAnimationFrame = function (callback) {
  setTimeout(callback, 0);
  return 0;
};

module.exports = React;
