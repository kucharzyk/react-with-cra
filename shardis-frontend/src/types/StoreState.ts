import { CounterState } from '../reducers/counter';

export default interface StoreState {
  counter: CounterState;
}
