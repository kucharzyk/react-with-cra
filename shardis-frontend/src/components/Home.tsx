import * as React from 'react';
import { StatelessComponent } from 'react';

const Home: StatelessComponent = () => (
  <div className="Home">
    <h2>This is Home component</h2>
  </div>
);

export default Home;
