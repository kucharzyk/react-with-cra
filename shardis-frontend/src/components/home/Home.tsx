import * as React from 'react';
import { StatelessComponent } from 'react';
import Counter from '../../containers/counter/Counter';

const Home: StatelessComponent = () => (
  <div className="Home">
    <h2>This is Home component</h2>
    <hr/>
    <Counter/>
  </div>
);

export default Home;
