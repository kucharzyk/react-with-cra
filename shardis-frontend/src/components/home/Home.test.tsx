import * as React from 'react';
import { StatelessComponent } from 'react';
import Home from './Home';
import StoreState from '../../types/StoreState';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { StaticRouter } from 'react-router';
import createShallow from 'material-ui/test-utils/createShallow';
import toJson from 'enzyme-to-json';

describe('Home', () => {

  let WrappedComponent: StatelessComponent;

  beforeEach(() => {
    const mockStore = configureStore<StoreState>();
    const store = mockStore({
      counter: 1
    });

    WrappedComponent = () => (
      <Provider store={store}>
        <StaticRouter location={'/'} context={{}}>
          <Home/>
        </StaticRouter>
      </Provider>
    );
  });

  it('renders snapshot', () => {
    const shallowComponent = createShallow({untilSelector: 'Home'})(<WrappedComponent/>);
    expect(toJson(shallowComponent)).toMatchSnapshot();
  });

});
