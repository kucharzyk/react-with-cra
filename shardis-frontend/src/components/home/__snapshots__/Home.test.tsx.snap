// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders snapshot 1`] = `
<div
  className="Home"
>
  <h2>
    This is Home component
  </h2>
  <hr />
  <Connect(Counter) />
</div>
`;
